src/test/scala/strawman/collection/test/Test.scala

```diff
@@ -288,6 +289,37 @@ class StrawmanTest {
     println(xs17.to(List))
   }
 
+  def sortedSets(xs: immutable.SortedSet[Int]): Unit = {
+    val xs1 = xs.map((x: Int) => x.toString)
+    val xs2: immutable.SortedSet[String] = xs1
+  }
+
+  def mapOps(xs: Map[Int, String]): Unit = {
+    val xs1 = xs.map((k, v) => (v, k))
+    val xs2: strawman.collection.Map[String, Int] = xs1
+    val xs3 = xs.map(kv => (kv._2, kv._1))
+    val xs4: strawman.collection.Iterable[(String, Int)] = xs3
```

Comment:
Note that here the distinction between the `map` overload that returns a `Map` and the one that returns an `Iterable` is based on the arity of the supplied function.

This is because `map` method of `Map` is defined as follows:

~~~ scala
def map[K2, V2](f: (K, V) => (K2, V2)): C[K2, V2]
~~~

We could alternatively define it as follows:

~~~ scala
def map[K2, V2](f: ((K, V)) => (K2, V2)): C[K2, V2]
~~~

Note that here,  `f` takes one parameter, which is a tuple, where in the first snippet, it takes two parameters.

I’m not sure which solution is better… I think both can work and make sense.

With the binary function I like the ability to just write `kvs.map((k, v) => …)` without having to write `{ case (k, v) => … }`. But note that dotty already makes it possible to write `kvs.map((k, v) => …)` even though `f` would have type `((K, V)) => (K2, V2)`. Maybe we could put this feature into scalac too…

Not sure I’m clear… Do you have any strong preference and why?